packages/devtools_app/lib/src/flutter/table.dart

```diff
@@ -699,15 +655,62 @@ class _TableState<T> extends State<_Table<T>> with AutoDisposeMixin {
       setState(() {
         final selection = oldWidget.selectionNotifier.value;
         if (selection.scrollIntoView) {
-          final newPosition = selection.displayRow * defaultRowHeight;
-          // TODO(terry): Optimize selecting row that's in the viewport
-          //              otherwise we jumpTo.
-          scrollController.jumpTo(newPosition);
+          final selectedDisplayRow = selectionRowNumber(selection.node);
+
+          // Compute total rows above immediate parent.
+          var parentNode = selection.node.parent;
+          if (parentNode.index != -1) {
+            parentNode = parentNode.parent;
+          }
+          while (parentNode.index != -1) {
+            parentNode = parentNode.parent;
+          }
+
+          final newPos = selectedDisplayRow * defaultRowHeight;
+          // TODO(terry): Optimize selecting row, if row's visible in
+          //              the viewport just select otherwise jumpTo row.
+          scrollController.jumpTo(newPos);
         }
       });
     });
   }
 
+  /// Return the number of visible rows above the selected node.
+  int selectionRowNumber(dynamic selectedNode) {
```

Comment:
why is selectedNode type `dynamic`?

Reply:
I'm adding a TODO that I need to fix immediately after committing this PR.  The selectionNotifier handling needs to be better refactored between _Table and TreeTable (TreeNode).  This works today because this code is only called when selection.scrollIntoView is true and that's only for TreeTable's.